### PR TITLE
feat: array flags trim whitespace

### PR DESCRIPTION
**Pull request by mshanemc**
per discussion with @amphro yesterday.

makes sfdxFlags friendlier so developers don't have to do this:
https://github.com/salesforcecli/plugin-user/blob/a01e5a9d57ce5b88f0f6a06dc441a72bf6225c25/src/commands/force/user/permset/assign.ts#L53

### CODE REVIEW

**Comment by peternhale** [comment]
@amphro @mshanemc, I don't see any tests that exercise array parsing with a custom delimiter.

Would it be possible to add test cases for delimiter as part of this work?

File: packages/command/src/sfdxFlags.ts [hunk 1]
```diff
@@ -228,7 +228,10 @@ function buildStringArray(kind: flags.Kind, options: flags.Array<string>): flags
   return option(kind, rest, (val) => {
     // eslint-disable-next-line no-useless-escape
     const regex = /\"(.*?)\"|\'(.*?)\'|,/g;
```
Comment (mshanemc) [inline packages/command/src/sfdxFlags.ts:230]: @peternhale it doesn't look like stringArray supported custom delimiters properly (adding a map option pushes it into the buildMappedArray method which does.

Comment (peternhale) [reply packages/command/src/sfdxFlags.ts:230]: The issue for me is there are no tests that create a array flag that defines a custom delimiter.

**Review by shetzel — APPROVED** [review]

File: packages/command/src/sfdxFlags.ts (hunk 1 above)
Comment (mshanemc) [reply packages/command/src/sfdxFlags.ts:230]: added the tests, and fixed the "works with maps AND string problems" over here
https://github.com/forcedotcom/cli-packages/pull/139

**Pull request merged**